### DISCUSSION TITLE
[JupyterLite] Using skimage, pandas, pooch: '_lzma' issue

### DISCUSSION THREAD

**Discussion by labdala** [pyodide/pyodide#2930] [Q&A] [answered]
Hi, I'm using trying to import a skimage dataset (data.cells3d()[20:]) to a JupyterLite notebook. To use this file, I need to install pooch. 
I am not sure if, at this point, pyodide supports this, but here is what I tried:
```
import piplite
piplite.install("matplotlib==3.3.3")
piplite.install("itkwidgets>=1.0a5")
piplite.install("skimage==0.19.2")
piplite.install("pandas>=1.4.0")
piplite.install("plotly")
piplite.install("pooch>=1.3.0")
```
and also tried this just in case:
```
import micropip
await micropip.install('https://files.pythonhosted.org/packages/8d/64/8e1bfeda3ba0f267b2d9a918e8ca51db8652d0e1a3412a5b3dbce85d90b6/pooch-1.6.0-py3-none-any.whl')
```
I have no problems importing pandas
```
import pandas as pd
```
but do get the  error `ModuleNotFoundError: No module named '_lzma'` when import pooch
```
import pooch
```
I've seen there are a couple discussions about lzma related to Pandas, [1](https://github.com/pyodide/pyodide/issues/1986), [2](https://github.com/jupyterlite/jupyterlite/issues/199), [3](https://github.com/pyodide/pyodide/issues/877), but I am still puzzled on whether this is a fixable issue.  Can anyone help me?

**Comment by rth** [comment, marked as answer]
The issue is that lzma module is not currently available in Pyodide (it's optional in CPython). So pooch should handle [here](https://github.com/fatiando/pooch/blob/ac1f19b1e5d6e01f1155525423f96ff04930b017/pooch/processors.py#L14) the case where the module cannot be imported (instead of assuming that the lzma module is always importable).

You can open an issue about this upsteam. A temporary workaround would be to remove that top-level import, create a new wheel that you put somewhere and install from there.

**Reply by labdala** [reply]
Got it, this is very helpful. Thank you so much @rth !

**Reply by hoodmane** [reply]
@rth Is there an issue open about `lzma`? What is our position on it again? That we want it but there is no emscripten port and we haven't gotten to it?

**Reply by rth** [reply]
Opened https://github.com/pyodide/pyodide/issues/2937

**Comment by ryanking13** [comment]
lzma is now included in the dev branch (#2939). But it is unvendored by default so you'll need to use ~~`micropip.install("_lzma")`.~~

_Edit:_

`micropip.install("lzma")`

**Reply by labdala** [reply]
Thank you so much @ryanking13

**Reply by hoodmane** [reply]
Now released in Pyodide v0.21.1
